src/main/kotlin/com/shortener/domain/UrlEntry.kt

```diff
@@ -0,0 +1,61 @@
+package com.shortener.domain
+
+import org.springframework.format.annotation.DateTimeFormat
+import java.time.LocalDateTime
+import javax.persistence.*
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Entity(name = "URL_ENTRIES")
+class UrlEntry {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    private var id: Long? = null
+
+    @Column(name = "long_url")
+    @Size(max = 2048)
+    @NotBlank
+    private var longUrl: String? = null
+
+    @Column(name = "created_at")
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    @NotNull
+    private var createdAt: LocalDateTime? = null
+
+    @Column(name = "expires_at")
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    @NotNull
+    private var expiresAt: LocalDateTime? = null
+
+    fun getId(): Long? {
```

Comment:
What's the point of private properties and adding getters and setters? This is just an anti-pattern that was common in java (and unfortunately I don't think you could have avoided it when using frameworks). I think JPA with kotlin works fine if you remove the private access modifier as well as getters and setters.

Reply:
P.S: for get you could have just inlined the functions.  fun getLongUrl() =  longUrl

Reply:
I'm into Kotlin for a few days only, still learning a lot of things. I don't know yet what is good or bad, or customary, or antipatterns or whatever, in this language. Asking "why's" and "what's" on Kotlin specific stuff will not get you answers from me. 

Suggestions are always welcomed.